lib/fs.js

```diff
@@ -1004,7 +1004,7 @@ fs.fchmodSync = function(fd, mode) {
   return binding.fchmod(fd, modeNum(mode));
 };
 
-if (constants.hasOwnProperty('O_SYMLINK')) {
+if ('O_SYMLINK' in constants) {
```

Comment:
As `constants` inherit from `null`, will the performance implication still be applicable?